src/librustc/ty/sty.rs

```diff
@@ -2402,17 +2402,84 @@ pub struct Const<'tcx> {
 static_assert_size!(Const<'_>, 48);
 
 impl<'tcx> Const<'tcx> {
+    /// Literals and const generic parameters are eagerly converted to a constant, everything else
+    /// becomes `Unevaluated`.
+    pub fn from_hir_anon_const(tcx: TyCtxt<'tcx>, def_id: DefId, ty: Ty<'tcx>) -> &'tcx Self {
+        debug!("Const::from_hir_anon_const(id={:?})", def_id);
+
+        let hir_id = tcx.hir().as_local_hir_id(def_id).unwrap();
+
+        let body_id = tcx.hir().body_owned_by(hir_id);
```

Comment:
If you want to limit this to `AnonConst`s, use `get` and support only ``hir::Node::AnonConst`, instead of relying on `body_owned_by`.